che-plugin-openshift-ext-client/src/main/java/org/eclipse/che/ide/ext/openshift/client/build/BuildsPresenter.java

```diff
@@ -0,0 +1,491 @@
+
+    private String webSocketBasePath = null;
+
+    private String oauthToken = null;
+
+    /** A set of build watchers. Use `namespace` for key */
+    private HashMap<String, BuildsWatcher> buildWatchers = new HashMap<>();
+
+    /** A set of logs watchers. Use `namespace` + `/` + `name` for key */
+    private HashMap<String, LogsWatcher> logsWatchers = new HashMap<>();
+
+    /** A set of notifications */
+    private HashMap<String, Notification> notifications = new HashMap<>();
+
+    @Inject
+    public BuildsPresenter(final BuildsView view,
+                           final WorkspaceAgent workspaceAgent,
+                           final OpenshiftWebSocketPathProvider wsPathProvider,
+                           final EventBus eventBus,
+                           final DtoFactory dtoFactory,
+                           final NotificationManager notificationManager,
+                           final OpenshiftLocalizationConstant locale,
+                           final ProjectServiceClient projectServiceClient) {
+        this.view = view;
+        this.workspaceAgent = workspaceAgent;
+        this.dtoFactory = dtoFactory;
+        this.notificationManager = notificationManager;
+        this.locale = locale;
+        this.projectServiceClient = projectServiceClient;
+        this.wsPathProvider = wsPathProvider;
+
+        view.setDelegate(this);
+
+        eventBus.addHandler(OAuthTokenChangedEvent.TYPE, this);
+    }
+
+    @Override
+    public void onOAuthTokenChanged(OAuthTokenChangedEvent event) {
+        oauthToken = event.getToken();
+
+        if (oauthToken == null) {
+            stopWatching();
+        } else {
+            if (WebSocket.isSupported()) {
+                getWebSocketBasePath();
+            }
+        }
+    }
+
+    /**
+     * Retrieves a base path for OpenShift WebSocket connections.
+     */
+    private void getWebSocketBasePath() {
+        wsPathProvider.get(new AsyncCallback<String>() {
+            @Override
+            public void onSuccess(String webSocketPath) {
+                webSocketBasePath = webSocketPath;
+                checkWorkspaceProjects();
+            }
+
+            @Override
+            public void onFailure(Throwable caught) {
+                Log.error(getClass(), caught.getMessage());
+            }
+        });
+    }
+
+    /**
+     * Checks workspace projects for being deployed on OpenShift and starts observation for builds for that projects.
+     */
+    private void checkWorkspaceProjects() {
+        projectServiceClient.getProjects(false).then(new Operation<List<ProjectConfigDto>>() {
+            @Override
+            public void apply(List<ProjectConfigDto> projects) throws OperationException {
+                for (ProjectConfigDto project : projects) {
+                    if (project.getMixins().contains("openshift") &&
+                            project.getAttributes().containsKey(OpenshiftProjectTypeConstants.OPENSHIFT_NAMESPACE_VARIABLE_NAME)) {
```

Comment:
It is required variable. So If project has mixin openshift it must have this variable